src/client/sandbox/storages/wrapper.ts

```diff
@@ -4,288 +4,227 @@ import { parseProxyUrl } from '../../utils/url';
 import * as destLocation from '../../utils/destination-location';
 import * as JSON from 'json-hammerhead';
 import nativeMethods from '../native-methods';
+import { StorageProxy } from '../../../typings/client';
 
-const API_KEY_PREFIX = 'hammerhead|api-key-prefix|';
-const KEY            = 0;
-const VALUE          = 1;
 
-function getWrapperMethods () {
-    const methods = [];
+const STORAGE_WRAPPER_KEY = 'hammerhead|get-storage-wrapper';
+const EMPTY_OLD_VALUE_ARG = isIE ? '' : null;
+const INTERNAL_WEAK_MAP   = new WeakMap<StorageWrapper, StorageInternalInfo>();
```

Comment:
Bad name.
Compare with `internal_number`, `internal_boolean`.
What about the `storageWrappersInternalInfo` name?